Source/SwiftLintFramework/Reporters/EmojiReporter.swift

```diff
@@ -0,0 +1,70 @@
+//
+//  EmojiReporter.swift
+//  SwiftLint
+//
+//  Created by Michał Kałużny on 01/12/2016.
+//  Copyright © 2016 Realm. All rights reserved.
+//
+
+import Foundation
+
+public struct EmojiReporter: Reporter {
+    public static let identifier = "emoji"
+    public static let isRealtime = false
+
+    public var description: String {
+        return "Reports violations in the format that's both fun and easy to read."
+    }
+
+    public static func generateReport(violations: [StyleViolation]) -> String {
+        return violations.categorise {
+                (violation) in
+                violation.location.file ?? "Other"
+            }.map {
+                (filename, violations) in
```

Comment:
closure parameters on the same line as opening brace please: `{ filename, violations in`